### PR TITLE
feat(uat): add core device discover to SDK-based client

### PR DESCRIPTION
**Pull request by bgklika**
**Issue #, if available:**
Implement discovery in SDK-based clients
Update "generate documentation" section in uat/Readme.md file

**Description of changes:**
- Extend protocol by new request
- Implement discover in SDK-based client only
- Update control to support new request
- Add new step to MqttControlSteps
- Add GGMQ-1-T103 scenario outline
- Update limitation block of uat/REAME.md
- Add "Generate documentation" block to REAME.md of mosquitto C client

**Why is this change necessary:**
Currently we have implemented discovery of core device broker in test framework but usually it done by MQTT client.
By this PR we add implementation of discover to SDK-based client.

**How was this change tested:**
GGMQ-1-T103 scenario outline has been added to test new feature.

**Any additional information or context required to review the change:**

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: uat/README.md [hunk 1]
```diff
@@ -177,7 +177,11 @@ mvn javadoc:javadoc
...
+See individual README.md files for Python and C clients.
+
 ## Limitations
 MQTT clients based on IoT Device SDK for Java v2, mosquitto C, Paho Java, Paho Python do no provide API to get information from PUBREC/PUBREL/PUBCOMP packages used when messages published with QoS 2.
 
-Not all features of MQTT v5.0 have been implemented in clients and are supported by gRPC proto and the control as was requested, these are not bugs but designed by requirement.
\ No newline at end of file
+Not all features of MQTT v5.0 have been implemented in clients and are supported by gRPC proto and the control as was requested, these are not bugs but designed by requirement.
+
+Discover of Core device broker feature is implemented only in the client based on AWS IoT device SDK library.
```
Comment (MikeDombo) [inline uat/README.md:187]: _Discovery_

File: uat/custom-components/client-java-sdk/src/main/java/com/aws/greengrass/testing/mqtt5/client/DiscoverClient.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+package com.aws.greengrass.testing.mqtt5.client;
+
+import com.aws.greengrass.testing.mqtt.client.CoreDeviceDiscoverReply;
+import com.aws.greengrass.testing.mqtt.client.CoreDeviceDiscoverRequest;
+import com.aws.greengrass.testing.mqtt5.client.exceptions.DiscoverException;
+
+/**
+ * Interface of discovery client.
+ */
+public interface DiscoverClient {
```
Comment (MikeDombo) [inline uat/custom-components/client-java-sdk/src/main/java/com/aws/greengrass/testing/mqtt5/client/DiscoverClient.java:15]: Discovery

File: uat/custom-components/client-java-sdk/src/main/java/com/aws/greengrass/testing/mqtt5/client/exceptions/DiscoverException.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.testing.mqtt5.client.exceptions;
+
+/**
+ * Client's exception related to discover parts.
+ */
+public class DiscoverException extends ClientException {
```
Comment (MikeDombo) [inline uat/custom-components/client-java-sdk/src/main/java/com/aws/greengrass/testing/mqtt5/client/exceptions/DiscoverException.java:11]: Discovery

File: uat/custom-components/client-java-sdk/src/main/java/com/aws/greengrass/testing/mqtt5/client/exceptions/DiscoverException.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.testing.mqtt5.client.exceptions;
+
+/**
+ * Client's exception related to discover parts.
```
Comment (MikeDombo) [inline uat/custom-components/client-java-sdk/src/main/java/com/aws/greengrass/testing/mqtt5/client/exceptions/DiscoverException.java:9]: Discovery

File: uat/README.md (hunk 1 above)
Comment (bgklika) [reply uat/README.md]: Fixed in 13f192dc2c5f293be1752b82b975e696a351858e

File: uat/custom-components/client-java-sdk/src/main/java/com/aws/greengrass/testing/mqtt5/client/exceptions/DiscoverException.java (hunk 3 above)
Comment (bgklika) [reply uat/custom-components/client-java-sdk/src/main/java/com/aws/greengrass/testing/mqtt5/client/exceptions/DiscoverException.java]: Fixed in 13f192dc2c5f293be1752b82b975e696a351858e

File: uat/custom-components/client-java-sdk/src/main/java/com/aws/greengrass/testing/mqtt5/client/exceptions/DiscoverException.java (hunk 4 above)
Comment (bgklika) [reply uat/custom-components/client-java-sdk/src/main/java/com/aws/greengrass/testing/mqtt5/client/exceptions/DiscoverException.java]: Fixed in 13f192dc2c5f293be1752b82b975e696a351858e

**Review by MikeDombo — APPROVED** [review]

**Pull request merged**